packages/experimental/kai/src/hooks/useAppRouter.ts

```diff
@@ -0,0 +1,105 @@
+//
+// Copyright 2023 DXOS.org
+//
+
+import { useEffect } from 'react';
+import { useNavigate, useParams } from 'react-router-dom';
+
+import { Invitation, InvitationEncoder, Space } from '@dxos/client';
+import { PublicKey } from '@dxos/keys';
+import { useClient, useIdentity, useSpaces } from '@dxos/react-client';
+
+import { oncePerWindow } from './once';
+import { defaultFrameId, FrameDef, useFrames } from './useFrames';
+// TODO(burdon): Create defs/helpers for other routes.
+export enum Section {
+  FRAME = 'frame',
+  REGISTRY = 'registry',
+  SETTINGS = 'settings'
+}
+
+const truncateKey = (key: PublicKey) => key.toHex().slice(0, 8);
+
+export const findSpace = (spaces: Space[], spaceKey: string): Space | undefined =>
+  spaces.find((space) => truncateKey(space.key) === spaceKey);
+
+const encodeFrame = (frame: string) => frame.replaceAll('.', '_');
+export const decodeFrame = (frame: string) => frame.replaceAll('_', '.');
+
+export const createPath = ({
+  spaceKey,
+  section,
+  frame,
+  objectId
+}: {
+  spaceKey?: PublicKey;
+  section?: string;
+  frame?: string;
+  objectId?: string;
```

Comment:
this should probably be renamed `documentId`, right?

Reply:
I ECHO Objects are by default Documents.